pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -349,9 +374,34 @@ func getOptimalBlockIOForDevice(path string) (*api.BlockIO, error) {
 			blockIO.LogicalBlockSize = uint(physicalSize)
 		}
 	}
+	if *blockIO.DiscardGranularity%blockIO.LogicalBlockSize != 0 {
+		log.Log.Infof("Invalid discard granularity %d. Matching it to physical size %d", *blockIO.DiscardGranularity, blockIO.PhysicalBlockSize)
+		blockIO.DiscardGranularity = pointer.P(uint(physicalSize))
+	}
 	return blockIO, nil
 }
 
+func getDiscardGranularity(safePath *safepath.Path) (uint64, error) {
+	fileInfo, err := safepath.StatAtNoFollow(safePath)
+	if err != nil {
+		return 0, fmt.Errorf("could not stat file %s. Reason: %w", safePath.String(), err)
+	}
+	stat := fileInfo.Sys().(*syscall.Stat_t)
+	major := unix.Major(stat.Rdev)
```

Comment:
I think you need to wrap the input to unix.Major for cross platform https://github.com/kubevirt/kubevirt/blob/d5045265c1e2c9deef921971b1a720d890577441/vendor/github.com/opencontainers/runc/libcontainer/devices/device_unix.go#L43